services/accumulo-service/src/main/java/datawave/microservice/accumulo/mock/MockAccumuloDataService.java

```diff
@@ -0,0 +1,99 @@
+package datawave.microservice.accumulo.mock;
+
+import com.google.common.base.Preconditions;
+import datawave.microservice.accumulo.lookup.LookupService;
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.BatchWriterConfig;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * When the <strong>mock</strong> profile is activated, this bean will automatically ingest a tiny amount of data into an in-memory accumulo instance. The
+ * tables and data here are designed mainly for this application's unit tests, but the service may also be useful for integration testing
+ * <p>
+ * TODO: Replace with a better mock data provider (i.e., a more generic service that supports DataWave's query use cases and underlying table schema)
+ */
+@Service
+@Profile("mock")
+public class MockAccumuloDataService {
```

Comment:
I wonder if this could be a CommandLineRunner instead. Seems like you're only after the side effect of creating the bean created, and that's what CommandLineRunner is suited for.

Reply:
Not sure I follow what you had in mind exactly. The intent is just auto-init an in-memory Accumulo with some dummy data by simply activating the mock profile, so that you can run the microservice without an actual Accumulo instance. I also inject this bean into my junits and make use of its public methods for various tests. If those use cases can be supported more cleanly with CommandLineRunner, then I'm open to it.

Reply:
FWIW, I tried modifying the class to implement `CommandLineRunner`, moved the existing `setupMockWarehouseTables` call from the constructor to the `run` method, and verified that everything still works as intended. Did you have something different in mind?

Reply:
Sorry, missed this... Not a big deal since it's covered by the mock profile, but I was just thinking having a CommandLineRunner might be more appropriate than an actual Service annotation. I didn't realize at the time I commented, that you were injecting this in tests.